easypy/logging.py

```diff
@@ -126,6 +126,47 @@ def get_console_handler():
             return handler
 
 
+class ThreadControl(logging.Filter):
+    """
+    Used by ContextLoggerMixin .solo and .suppressed methods to control logging to console
+    To use, add it to the logging configuration as a filter in the console handler
+
+        ...
+        'filters': {
+            'thread_control': {
+                '()': 'easypy.logging.ThreadControl'
+            }
+        },
+        'handlers': {
+            'console': {
+                'class': 'logging.StreamHandler',
+                'filters': ['thread_control'],
+            },
+
+    """
+
+    CONTEXT = ThreadContexts(counters='silenced')
+    SELECTED = [None]
+    LOCK = threading.RLock()
+
+    @classmethod
+    @contextmanager
+    def solo(cls):
+        try:
+            with cls.LOCK:
+                cls.SELECTED.append(threading.current_thread())
+            yield
+        finally:
+            with cls.LOCK:
+                cls.SELECTED.pop(-1)
```

Comment:
I'm not that familiar with the logging configuration, but is it safe to do something like this to a list shared between threads?